### PR TITLE
documentation: add UML for test_flows refactoring

### PR DESCRIPTION
**Pull request by arnout**

### CODE REVIEW

**Comment by arnout** [comment]
The UML is not very complete, it's more to give an overview of the direction in which we want to go.

It needs to be adapted to match with the [boardfarm](https://github.com/mattsm/boardfarm/) architecture. Since @abelog will analyse that (#919) I've added him as assignee to this PR.

File: documentation/images/plantuml/test_flows.iuml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+class ClientSteeringTest {
+   test_client_association()
+   test_client_steering_dummy()
+   test_optimal_path_dummy()
+}
+
+FlowTest <|-- ClientSteeringTest
+
+@enduml
```
Comment (tomereli) [inline documentation/images/plantuml/test_flows.iuml]: Missing line break at end of file

**Review by tomereli — APPROVED** [review]

**Comment by morantr** [comment]
I generated the UML and this is what I've got.
image

**Review by morantr — APPROVED** [review]

**Comment by arnout** [comment]
In the mean time, I've changed the UML a little to correspond better with what is actually there.

Therefore, I'll include this as part of #1031

**Pull request closed without merging**